web3.js/src/transaction.js

```diff
@@ -432,37 +444,46 @@ export class Transaction {
    * Verify signatures of a complete, signed Transaction
    */
   verifySignatures(): boolean {
-    return this._verifySignatures(this.serializeMessage());
+    return this._verifySignatures(this.serializeMessage(), true);
   }
 
   /**
    * @private
    */
-  _verifySignatures(signData: Buffer): boolean {
-    let verified = true;
+  _verifySignatures(signData: Buffer, requireAllSignatures: boolean): boolean {
     for (const {signature, publicKey} of this.signatures) {
+      if (signature === null) {
+        return !requireAllSignatures;
+      }
       if (
         !nacl.sign.detached.verify(signData, signature, publicKey.toBuffer())
       ) {
-        verified = false;
+        return false;
       }
     }
-    return verified;
+    return true;
   }
 
   /**
    * Serialize the Transaction in the wire format.
-   *
-   * The Transaction must have a valid `signature` before invoking this method
    */
-  serialize(): Buffer {
+  serialize(config?: SerializeConfig): Buffer {
     const {signatures} = this;
-    if (!signatures || signatures.length === 0) {
```

Comment:
Checking for `!signatures` appears to be unnecessary.  `signatures` is always an array AFAICT